qcodes/dataset/data_set.py

```diff
@@ -1353,7 +1354,7 @@ def _finalize_res_dict_numeric_text_or_complex(
                     flat_results[dep.name] = result_dict[dep].ravel()
             for inff in inff_params:
                 if numpy.shape(result_dict[inff]) == ():
-                    flat_results[inff.name] = numpy.repeat(result_dict[dep], N)
```

Comment:
This looks like a bug in the original code but also looks like we have no tests for this